docs/payload_context.rst

```diff
@@ -0,0 +1,84 @@
+.. _payload_context:
+
+Payload Context
+===============
+
+Sometimes it is beneficial to pass some contextual information from the trigger
+to the trigger listener along the payload. Examples are:
+
+- tracing information that allows to track complex request processing in a
+  multi component system
+- in a multitenant system a tenant information to be able to identify the
+  tenant that peformed an operation that triggered a notification
+
+
+This can be done by using **Payload Context**. This feature includes:
+
+- ability to add an additional information to the payload in the trigger
+- ability to filter by the fields in the context in the listener process
+- ability to use ``context`` fields in the listener callbacks
+
+
+Add ``context`` to payload in the trigger
+-----------------------------------------
+
+Before doing updates that produce notifications set the context that should be
+passed using ``pgpubsub.set_notification_context`` function. 
+
+.. code-block:: python
+
+    from pgpubsub import set_notification_context
+
+    set_notification_context({'some-key': 'some-value'})
+
+The setting is effective till the connection is closed. Alternatively the setting
+``PGPUBSUB_TX_BOUND_NOTIFICATION_CONTEXT`` can be used to clean the context at the end 
```

Comment:
I don't fully understand the naming `PGPUBSUB_TX_BOUND_NOTIFICATION_CONTEXT` here. What does "TX_BOUND" mean? Should we also mention what kind of values this setting can take?

Reply:
TX is common abbreviation for `transaction`, I can rename this to be `PGPUBSUB_NOTIFICATION_CONTEXT_BOUND_TO_TRANSACTION` if you think this is better.